src/player.cpp

```diff
@@ -564,6 +579,65 @@ void Player::removeSkillTries(skills_t skill, uint64_t count, bool notify /* = f
 	}
 }
 
+void Player::setLoyaltySkill(uint8_t skill, uint64_t count)
+{
+	uint64_t currReqTries = vocation->getReqSkillTries(skill, skills[skill].level);
+	uint64_t nextReqTries = vocation->getReqSkillTries(skill, skills[skill].level + 1);
+	if (currReqTries >= nextReqTries) {
+		// player has reached max skill
+		return;
+	}
+
+	while ((skills[skill].bTries + count) >= nextReqTries) {
+		count -= nextReqTries - skills[skill].bTries;
+
+		skills[skill].bLevel++;
+		skills[skill].bTries = 0;
+
+		currReqTries = nextReqTries;
+		nextReqTries = vocation->getReqSkillTries(skill, skills[skill].bLevel + 1);
+		if (currReqTries >= nextReqTries) {
+			count = 0;
+			break;
+		}
+	}
+
+	skills[skill].percent = Player::getPercentLevel(count, nextReqTries);
+	skills[skill].bTries += count;
+}
+
+uint64_t Player::getAccumulatedSkillTries(uint8_t skill) const
+{
+	if (skill > SKILL_LAST) {
+		return 0;
+	}
+
+	uint64_t count = 0;
+	for (uint16_t level = MINIMUM_SKILL_LEVEL; level <= skills[skill].level; level++) {
+		count += vocation->getReqSkillTries(skill, level);
+	}
+
+	count += skills[skill].tries;
+
+	return count;
+}
+
+void Player::updateLoyaltySkill(uint8_t skill)
+{
+	if (skill > SKILL_LAST) {
+		return;
+	}
+
+	skills[skill].bLevel = skills[skill].level;
+	skills[skill].bTries = skills[skill].tries;
```

Comment:
is bLevel supposed to be baseLevel?
If so what do you think about just naming it baseLevel, baseTries and so on